editor/editor_node.cpp

```diff
@@ -698,21 +698,19 @@ void EditorNode::_notification(int p_what) {
 				scene_tabs->update_scene_tabs();
 			}
 
-			// Update the animation frame of the update spinner.
-			uint64_t frame = Engine::get_singleton()->get_frames_drawn();
-			uint64_t tick = OS::get_singleton()->get_ticks_msec();
-
-			if (frame != update_spinner_step_frame && (tick - update_spinner_step_msec) > (1000 / 8)) {
-				update_spinner_step++;
-				if (update_spinner_step >= 8) {
-					update_spinner_step = 0;
-				}
-
-				update_spinner_step_msec = tick;
-				update_spinner_step_frame = frame + 1;
+			if (update_spinner->is_visible()) {
+				// Update the animation frame of the update spinner.
+				uint64_t frame = Engine::get_singleton()->get_frames_drawn();
+				uint64_t tick = OS::get_singleton()->get_ticks_msec();
+
+				if (frame != update_spinner_step_frame && (tick - update_spinner_step_msec) > (1000 / 8)) {
+					update_spinner_step++;
+					if (update_spinner_step >= 8) {
+						update_spinner_step = 0;
+					}
 
-				// Update the icon itself only when the spinner is visible.
-				if (_should_display_update_spinner()) {
```

Comment:
The `_should_display_update_spinner()` call appears to do some relevant stuff. Wouldn't it make more sense to change the first check to `if (update_spinner->is_visible() and __should_display_update_spinner())`?

Reply:
Edit: Nevermind, the visibility of the spinner is decided by this method. But then the method itself is not needed anymore, it's only used in one place now.